src/PIL/JpegImagePlugin.py

```diff
@@ -158,7 +158,7 @@ def APP(self, marker):
     # If DPI isn't in JPEG header, fetch from EXIF
     if "dpi" not in self.info and "exif" in self.info:
         try:
-            exif = self._getexif()
+            exif = self.getexif()
```

Comment:
The main difference is that `getexif` from `Image` class does not cache the parsed Exif tags.
So this call is done and forget when the user will need it.

This is not what we should expect.